test/framework/src/main/java/org/elasticsearch/test/rest/client/RestTestClient.java

```diff
@@ -296,7 +303,17 @@ private static RestClient createRestClient(URL[] urls, Settings settings) throws
                 .register("http", PlainConnectionSocketFactory.getSocketFactory())
                 .register("https", sslsf)
                 .build();
-        CloseableHttpClient httpClient = RestClient.Builder.createDefaultHttpClient(socketFactoryRegistry);
+        PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+        //default settings may be too constraining
+        connectionManager.setDefaultMaxPerRoute(10);
+        connectionManager.setMaxTotal(30);
+
+        //default timeouts are all infinite
+        RequestConfig requestConfig = RequestConfig.custom().setConnectTimeout(CONNECT_TIMEOUT_MILLIS)
+            .setSocketTimeout(SOCKET_TIMEOUT_MILLIS)
+            .setConnectionRequestTimeout(CONNECTION_REQUEST_TIMEOUT_MILLIS).build();
+        CloseableHttpClient httpClient = HttpClientBuilder.create()
+            .setConnectionManager(connectionManager).setDefaultRequestConfig(requestConfig).build();
```

Comment:
ok I missed that. I wouldn't increase just everything here. Connection request timeout is about retrieving tcp connection from the internal pool that the http client has, I wouldn't touch that.
As for for connect timeout, not sure either, especially because we are not hitting a connection timeout but a socket timeout in our tests.

I don't like the fact that we have to copy the whole createDefaultHttpClient method here. But this is a usability concern that we can address in a separate issue.